octodns/manager.py

```diff
@@ -273,7 +273,7 @@ def sync(self, eligible_zones=[], eligible_targets=[], dry_run=True,
             for target, plan in plans:
                 plan.raise_if_unsafe()
 
-        if dry_run or config.get('always-dry-run', False):
+        if dry_run or self.config.get('always-dry-run', False):
```

Comment:
This is definitely a bug. Adding `self.` will prevent the incorrect access, but it'll actually get things looking into the wrong place for `always_dry_run`. That's supposed to be a per-zone config that prevents changes from being made to that zone. Looks like some refactoring has happened since the last time I tested this specific path. Probably the first thing to do is add a test case for it so that it doesn't happen again and the correct fix can be put in place. 

I'll have to look at it closer, but I think a bit more involved changes may be necessary. With how that function is structured now it may require adding a dry_run flag to the plan so that it can be skipped when encountered rather than aborting the whole sync. 

It looks like this path hasn't been tested since a long time ago when `sync` was per-zone :frowning_face:. Thanks for catching it. I think you may have b/c of an ~edge case where your zone didn't have the trailing `.` so the `config` var in the foreach up above was never set. 

`octodns-sync --debug --config-file=./config/production.yaml --doit --target pdns test.it`

vs 

`octodns-sync --debug --config-file=./config/production.yaml --doit --target pdns test.it.`